internal/server/data/migrations.go

```diff
@@ -594,7 +596,124 @@ func setDefaultOrgID() *migrator.Migration {
+						provider_group_id bigint NOT NULL,
+						group_id bigint NOT NULL
+					);
+					CREATE UNIQUE INDEX idx_org_provider_group_names ON provider_groups(organization_id, provider_id, name);
+			`)
+			if err != nil {
+				return fmt.Errorf("create provider groups table: %w", err)
+			}
 
+			if migrator.HasColumn(tx, "provider_users", "groups") {
+				// move the old data into the new tables
+				stmt := `
+					SELECT identity_id, provider_id, groups, identities.organization_id 
+					FROM provider_users
+					JOIN identities ON identities.id = provider_users.identity_id
+					WHERE groups != ''
+				`
+				rows, err := tx.Query(stmt)
+				if err != nil {
+					return fmt.Errorf("migrate existing provider groups: %w", err)
+				}
+
+				type groupProviderUser struct {
+					IdentityID uid.ID
+					ProviderID uid.ID
+					Groups     models.CommaSeparatedStrings
+					OrgID      uid.ID
+				}
+				for rows.Next() {
+					var groupProviderUser groupProviderUser
+					if err := rows.Scan(&groupProviderUser.IdentityID, &groupProviderUser.ProviderID, &groupProviderUser.Groups, &groupProviderUser.OrgID); err != nil {
+						return err
+					}
+
+					for _, name := range groupProviderUser.Groups {
+						// check if this provider user group exists yet
+						var providerGroupID uid.ID
+						err := tx.QueryRow(`SELECT id FROM provider_groups WHERE name = ? AND provider_id = ?`, name, groupProviderUser.ProviderID).Scan(&providerGroupID)
+						if err != nil && !strings.Contains(err.Error(), "no rows in result set") {
+							return err
+						}
+
+						if providerGroupID == 0 {
+							// get the corresponding infra group which already exists
+							var infraGroupID uid.ID
+							err := tx.QueryRow(`SELECT id FROM groups WHERE name = ? AND organization_id = ?`, name, groupProviderUser.OrgID).Scan(&infraGroupID)
+							if err != nil {
+								return err
+							}
+
+							// create the provider group, the infra group already exists
+							createPGStmt := `
+								INSERT INTO provider_groups(id, created_at, updated_at, provider_id, name, organization_id)
+								VALUES (?, ?, ?, ?, ?, ?)
+							`
+							providerGroupID = uid.New()
+							_, err = tx.Exec(createPGStmt, providerGroupID, time.Now(), time.Now(), groupProviderUser.ProviderID, name, groupProviderUser.OrgID)
+							if err != nil {
+								return err
+							}
+
+							// link this provider group to the infra group with the same name
+							_, err = tx.Exec(`
+								INSERT INTO provider_groups_groups ("provider_group_id", "group_id")
+								VALUES (?, ?) 
+								ON CONFLICT DO NOTHING
+								`,
+								providerGroupID, infraGroupID)
+							if err != nil {
+								return fmt.Errorf("map provider group to infra group: %w", err)
+							}
+						}
+						// link this provider user as a provider group member
+						_, err = tx.Exec(`
+							INSERT INTO provider_groups_identities ("provider_group_id", "identity_id")
```

Comment:
yeah I definitely find this weird. this should be a link between provider_group and provider_user, so provider_groups_provider_users. If we want this user to be a member of our infra group we need them also in groups_identities.